searchlib/src/vespa/searchlib/features/closest_feature.cpp

```diff
@@ -0,0 +1,288 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "closest_feature.h"
+#include "constant_tensor_executor.h"
+#include "distance_calculator_bundle.h"
+#include "valuefeature.h"
+#include <vespa/eval/eval/fast_value.h>
+#include <vespa/eval/eval/value_codec.h>
+#include <vespa/searchcommon/common/schema.h>
+#include <vespa/searchlib/fef/indexproperties.h>
+#include <vespa/searchlib/fef/parameterdescriptions.h>
+#include <vespa/searchlib/fef/test/dummy_dependency_handler.h>
+#include <vespa/searchlib/tensor/distance_calculator.h>
+#include <vespa/searchlib/tensor/fast_value_view.h>
+#include <vespa/searchlib/tensor/i_tensor_attribute.h>
+#include <vespa/searchlib/tensor/serialized_tensor_ref.h>
+#include <vespa/searchlib/tensor/subspace_type.h>
+#include <vespa/vespalib/util/stash.h>
+
+#include <vespa/log/log.h>
+LOG_SETUP(".features.closest_feature");
+
+using search::fef::FeatureType;
+using search::fef::FieldInfo;
+using search::fef::ParameterDataTypeSet;
+using search::tensor::FastValueView;
+using search::tensor::ITensorAttribute;
+using search::tensor::SubspaceType;
+using search::tensor::VectorBundle;
+using vespalib::eval::CellType;
+using vespalib::eval::FastValueBuilderFactory;
+using vespalib::eval::TypedCells;
+using vespalib::eval::TypifyCellType;
+using vespalib::eval::Value;
+using vespalib::eval::ValueType;
+using vespalib::string_id;
+using vespalib::typify_invoke;
+
+using namespace search::fef::indexproperties;
+
+namespace {
+
+struct SetIdentity {
+    template <typename T>
+    static void invoke(void *space, size_t size) {
+        assert(size == sizeof(T));
+        *(T *) space = 1.0;
+    }
+};
+
+void setup_identity_cells(const ValueType& type, std::vector<char>& space, TypedCells& cells)
+{
+    if (type.is_double()) {
+        return;
+    }
+    space.resize(vespalib::eval::CellTypeUtils::mem_size(type.cell_type(), 1));
+    cells = TypedCells(space.data(), type.cell_type(), 1);
+    typify_invoke<1,TypifyCellType,SetIdentity>(type.cell_type(), space.data(), space.size());
+}
+
+}
+
+namespace search::features {
+
+/**
+ * Implements the executor for the closest feature.
```

Comment:
Consider removing this comment and instead add comments to ClosestSerializedExecutor and ClosestDirectExecutor, making it clear what is the difference between them and when they are used.